language-plutus-core/src/Language/PlutusCore/Rename/Internal.hs

```diff
@@ -121,27 +33,17 @@ withFreshenedTyVarDecl (TyVarDecl ann name kind) cont =
 -- to bring several term and type variables in scope before renaming the types of term variables.
 -- This situation arises when we want to rename a bunch of mutually recursive bindings.
 withFreshenedVarDecl
-    :: (HasUnique (tyname ann) TypeUnique, HasUnique (name ann) TermUnique)
+    :: (HasUniques (Term tyname name ann), MonadQuote m)
     => VarDecl tyname name ann
-    -> (ScopedRenameM (VarDecl tyname name ann) -> ScopedRenameM c)
-    -> ScopedRenameM c
+    -> (ScopedRenameT m (VarDecl tyname name ann) -> ScopedRenameT m c)
+    -> ScopedRenameT m c
 withFreshenedVarDecl (VarDecl ann name ty) cont =
     withFreshenedName name $ \nameFr -> cont $ VarDecl ann nameFr <$> renameTypeM ty
 
--- | Rename a name that has a unique inside.
-renameNameM
-    :: (HasUniquesRenaming renaming unique, HasUnique name unique)
-    => name -> RenameM renaming name
-renameNameM name = do
-    mayUniqNew <- lookupNameM name
-    pure $ case mayUniqNew of
-        Nothing      -> name
-        Just uniqNew -> name & unique .~ uniqNew
-
 -- | Rename a 'Type' in the 'RenameM' monad.
 renameTypeM
-    :: (HasUniquesRenaming renaming TypeUnique, HasUnique (tyname ann) TypeUnique)
-    => Type tyname ann -> RenameM renaming (Type tyname ann)
+    :: (HasRenaming ren TypeUnique, HasUniques (Type tyname ann), MonadQuote m)
+    => Type tyname ann -> RenameT ren m (Type tyname ann)
```

Comment:
Generalized `RenameM` to be a monad transformer, which allows to instantiate `RenameT` both for the renamer and for equality checks.